docs/development/getting-started.rst

```diff
@@ -0,0 +1,87 @@
+***********
+Development
+***********
+
+To set up the your development enviroment you will need to first install project deps:
+
+Linux:
+######
+.. code-block:: bash
+
+   Run the env install script: sudo sh scripts/libgit2.sh
+   pip install -r requirements.txt
+   pip install -r test-requirements.txt
+
+OS X:
+#####
+.. code-block:: bash
+
+   brew install libgit2
+   pip install pygit2==0.25.0
+   pip install -r requirements.txt
+   pip install -r test-requirements.txt
+
+
+
+.. note:: this will install latest libgit2 library so you have to make sure you install the same version library with pip ( current version: 0.25.0 )
+
+
+Virtualenv
+##########
+
+To use VirtualEnv we will need to add some extra steps
+
+1. virtualenv venv
+2. source ./venv/bin/activate
+3. export LIBGIT2=$VIRTUAL_ENV
+4. run modified bash below
+
+
+.. code-block:: bash
+
+    #!/bin/sh
+
+    # Ubuntu 16.04 Install only
+
+    sudo apt install git cmake make -y
+    sudo apt-get install -y python-dev libffi-dev libssl-dev libxml2-dev libxslt1-dev libssh2-1 libgit2-dev python-pip libgit2-24
+    sudo apt-get install -y pkg-config libssh2-1-dev libhttp-parser-dev libssl-dev libz-dev
+
+    LIBGIT_VERSION='0.25.0'
+
+    wget https://github.com/libgit2/libgit2/archive/v${LIBGIT_VERSION}.tar.gz
+    tar xzf v${LIBGIT_VERSION}.tar.gz
+    cd libgit2-${LIBGIT_VERSION}/
+    cmake . -DCMAKE_INSTALL_PREFIX=$LIBGIT2
+    make
+    sudo make install
+    export LDFLAGS="-Wl,-rpath='$LIBGIT2/lib',--enable-new-dtags $LDFLAGS"
+    sudo pip install pygit2==0.25.0
+    sudo ldconfig
+
+Test that it worked with:
+
+.. code-block:: bash
+
+   python -c 'import pygit2'
+
+
+Kubernetes
+##########
+
+To test your armada fixes/features you will need to set-up a Kubernetes cluster. We recommend:
+
+`Minikube <https://github.com/kubernetes/minikube#installation>`_
+
+`Halcyon <https://github.com/att-comdev/halcyon-vagrant-kubernetes>`_
+
+.. note:: When using Halcyon it will not generate a config file. Run the following commands to create one:
+
+.. code-block:: bash
+
+   kubectl config set-credentials myself --username=admin --password=secret
+   kubectl config set-cluster local-server --server=http://localhost:8080
+   kubectl config set-context default-context --cluster=local-server --user=myself
+   kubectl config use-context default-context
+   kubectl config set contexts.default-context.namespace the-right-prefix
+   kubectl config view
```

Comment:
[There is a mechanism for Halcyon to provide credentials](https://github.com/att-comdev/halcyon-vagrant-kubernetes/blob/master/get-k8s-creds.sh).